[xdoctest] reformat example code with google style in No.36-43

yuchen202 opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
Others

### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->
Others

### Description
<!-- Describe what you’ve done -->

修改如下文件的示例代码，使其通过 `xdoctest` 检查：



### Related links
 
- #55629
- #55295

yuchen202 committed:
赛题五：将 xdoctest 引入到飞桨框架工作流中

yuchen202 committed:
[xdoctest] Modify it to meet the requirements

力所能及地改了一些

megemini commented:
把标题改一下～

另外，这个 PR 里面文件太多了 ～ 🤣

看看能不能拆一下，或者先改一个 ～

yuchen202 committed:
再次修改

megemini commented:
标题还是要改一下，参考：
https://github.com/PaddlePaddle/Paddle/pull/55834

标题统一格式，好方便以后管理、搜索之类的 ～

yuchen202 committed:
这个是之前本地备份的时候多的文件，要删去

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -2081,67 +2082,64 @@ class centers and the shape of sampled_class_center will be [num_positive_class_
+        >>> num_classes = 20
+        >>> batch_size = 10
+        >>> num_samples = 6
+        >>> label = paddle.randint(low=0, high=num_classes, shape=[batch_size], dtype='int64')
```
- `paddle.randint` 也是带有随机性的，加一个随机种子吧

```suggestion
        >>> paddle.seed(2023)
        >>> label = paddle.randint(low=0, high=num_classes, shape=[batch_size], dtype='int64')
```

- 下面的输出，也记得按照seed的结果同步修改

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -2081,67 +2082,64 @@ class centers and the shape of sampled_class_center will be [num_positive_class_
-        #       [6 , 11, 10, 7 , 4 , 8 , 12, 12, 11, 12, 13, 1 , 3 , 9 , 7 , 9 , 4 , 6 , 0 , 2 ])
-        #Tensor(shape=[7], dtype=int64, place=CUDAPlace(1), stop_gradient=True,
-        #       [0, 1, 2, 3, 5, 7, 8])
+        >>> # required: distributed
+        >>> # Multi GPU, test_class_center_sample.py
```
好像依赖外部文件 `test_class_center_sample.py`，这段代码skip吧

```suggestion
        >>> # doctest: +SKIP('Depends on external files.')
        >>> # required: distributed
        >>> # Multi GPU, test_class_center_sample.py
```

SigureMo replied:
这里改用 `# doctest: +REQUIRES(env:DISTRIBUTED)` 即可，因为我们 CI 没有分布式环境，使用该 requires 就是自动跳过

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -2081,67 +2082,64 @@ class centers and the shape of sampled_class_center will be [num_positive_class_
+        >>> label = paddle.randint(low=0, high=num_classes, shape=[batch_size], dtype='int64')
+        >>> remapped_label, sampled_class_index = paddle.nn.functional.class_center_sample(label, num_classes, num_samples)
+
+        >>> print(label)
+        >>> print(remapped_label)
+        >>> print(sampled_class_index)
+        >>> # the output is
+        Tensor(shape=[10], dtype=int64, place=Place(cpu), stop_gradient=True,
+        [10, 3 , 11, 14, 15, 15, 7 , 12, 5 , 13])
+        Tensor(shape=[10], dtype=int64, place=Place(cpu), stop_gradient=True,
+        [3, 0, 4, 7, 8, 8, 2, 5, 1, 6])
+        Tensor(shape=[9], dtype=int64, place=Place(cpu), stop_gradient=True,
+        [3 , 5 , 7 , 10, 11, 12, 13, 14, 15])
```
这个形式看起来不太整洁，要不就一行print一行输出（输出的结果务必按照seed结果同步修改），如下：
```suggestion
        >>> print(label)
        Tensor(shape=[10], dtype=int64, place=Place(cpu), stop_gradient=True,
        [10, 3 , 11, 14, 15, 15, 7 , 12, 5 , 13])
        
        >>> print(remapped_label)
        Tensor(shape=[10], dtype=int64, place=Place(cpu), stop_gradient=True,
        [3, 0, 4, 7, 8, 8, 2, 5, 1, 6])
        
        >>> print(sampled_class_index)
        Tensor(shape=[9], dtype=int64, place=Place(cpu), stop_gradient=True,
        [3 , 5 , 7 , 10, 11, 12, 13, 14, 15])
```

sunzhongkai588 commented on python/paddle/nn/functional/activation.py:
```diff
@@ -1727,18 +1790,19 @@ def gumbel_softmax(x, temperature=1.0, hard=False, axis=-1, name=None):
+            >>> import paddle
+            >>> import paddle.nn.functional as F
+
+            >>> logits = paddle.randn([4, 6])
```
`randn`有随机性，加seed
```suggestion
            >>> paddle.seed(2023)
            >>> logits = paddle.randn([4, 6])
```
下面的输出也按照seed结果同步修改

sunzhongkai588 commented on python/paddle/nn/functional/activation.py:
```diff
@@ -1727,18 +1790,19 @@ def gumbel_softmax(x, temperature=1.0, hard=False, axis=-1, name=None):
+            >>> temperature = 0.01
+            >>> gumbel_softmax = F.gumbel_softmax(logits, temperature)
+            >>> print(gumbel_softmax)
+            >>> #out's value is as follows:
```
这行有点多余，删了吧
```suggestion

```

sunzhongkai588 commented on python/paddle/nn/functional/activation.py:
```diff
@@ -1727,18 +1790,19 @@ def gumbel_softmax(x, temperature=1.0, hard=False, axis=-1, name=None):
+            >>> gumbel_softmax = F.gumbel_softmax(logits, temperature)
+            >>> print(gumbel_softmax)
+            >>> #out's value is as follows:
+            Tensor(shape=[4, 6], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[0.00000000, 0.00000000, 1.        , 0.00000000, 0.00000000, 0.00000000],
+             [0.00000000, 0.00000000, 0.00000000, 0.00000000, 1.        , 0.00000000],
+             [0.00000000, 0.00000000, 1.        , 0.00000000, 0.00000000, 0.00000000],
+             [1.        , 0.00000000, 0.00000000, 0.00000000, 0.00000000, 0.00000000]])
```
注意修改输出结果

sunzhongkai588 commented on python/paddle/nn/functional/activation.py:
```diff
@@ -634,24 +658,24 @@ def rrelu(x, lower=1.0 / 8.0, upper=1.0 / 3.0, training=True, name=None):
+            ...                                   [[ 1.0, -2.0, -3.0,  4.0],
+            ...                                    [-5.0,  6.0,  7.0, -8.0],
+            ...                                    [ 6.0,  7.0,  8.0,  9.0]]]], dtype='float32')
+            >>> out = F.rrelu(input_tensor, 0.1, 0.3)
```
`rrelu`这个api也带有随机性，参考[rrelu API文档](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/rrelu_cn.html#rrelu)

- 加seed
- 下面的输出同步修改

```suggestion
            >>> paddle.seed(2023)
            >>> out = F.rrelu(input_tensor, 0.1, 0.3)
```

sunzhongkai588 commented on python/paddle/nn/functional/activation.py:
```diff
@@ -1655,17 +1719,16 @@ def glu(x, axis=-1, name=None):
+            ...         [-1.94900405, -1.89956081,  0.17134808, -1.11280477]]
+            ... )
+            >>> print(F.glu(x))
+            Tensor(shape=[2, 2], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[-0.15216254, -0.90048921],
+            [-1.05778778, -0.46985325]])
```
输出对齐，美观一点
```suggestion
            Tensor(shape=[2, 2], dtype=float32, place=Place(cpu), stop_gradient=True,
            [[-0.15216254, -0.90048921],
             [-1.05778778, -0.46985325]])
```

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -1060,39 +1056,38 @@ def dropout(
+            >>> y_1 = paddle.nn.functional.dropout(x, axis=1)
+            >>> y_01 = paddle.nn.functional.dropout(x, axis=[0,1])
+            >>> print(x)
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[1., 2., 3.],
+            [4., 5., 6.]])
```
输出对齐，美观一点
```suggestion
            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
            [[1., 2., 3.],
             [4., 5., 6.]])
```

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -1060,39 +1056,38 @@ def dropout(
+            [[1., 2., 3.],
+            [4., 5., 6.]])
+            >>> print(y_train)
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[2., 4., 0.],
+            [8., 0., 0.]])
```
```suggestion
            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
            [[2., 4., 0.],
             [8., 0., 0.]])
```

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -1060,39 +1056,38 @@ def dropout(
+            [[2., 4., 0.],
+            [8., 0., 0.]])
+            >>> print(y_test)
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[1., 2., 3.],
+            [4., 5., 6.]])
```
```suggestion
            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
            [[1., 2., 3.],
             [4., 5., 6.]])
```

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -1060,39 +1056,38 @@ def dropout(
+            [[1., 2., 3.],
+            [4., 5., 6.]])
+            >>> print(y_0)
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[2., 4., 6.],
+            [8. , 10., 12.]])
```
```suggestion
            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
            [[2., 4., 6.],
             [8. , 10., 12.]])
```

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -1060,39 +1056,38 @@ def dropout(
+            [[2., 4., 6.],
+            [8. , 10., 12.]])
+            >>> print(y_1)
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[2. , 4. , 6. ],
+            [8. , 10., 12.]])
```
```suggestion
            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
            [[2. , 4. , 6. ],
             [8. , 10., 12.]])
```

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -1060,39 +1056,38 @@ def dropout(
+            [[2. , 4. , 6. ],
+            [8. , 10., 12.]])
+            >>> print(y_01)
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[0., 0., 6.],
+            [0., 0., 0.]])
```
```suggestion
            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
            [[0., 0., 6.],
             [0., 0., 0.]])
```

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -1821,21 +1822,21 @@ def linear(x, weight, bias=None, name=None):
-          #     [2.1077576  2.1077576  2.1077576  2.1077576 ]]
+            >>> import paddle
+
+            >>> x = paddle.randn((3, 2), dtype="float32")
+            >>> # x: [[-0.32342386 -1.200079  ]
+            >>> #     [ 0.7979031  -0.90978354]
+            >>> #     [ 0.40597573  1.8095392 ]]
+            >>> weight = paddle.full(shape=[2, 4], fill_value="0.5", dtype="float32", name="weight")
+            >>> # weight: [[0.5 0.5 0.5 0.5]
+            >>> #          [0.5 0.5 0.5 0.5]]
+            >>> bias = paddle.ones(shape=[4], dtype="float32", name="bias")
+            >>> # bias: [1. 1. 1. 1.]
+            >>> y = paddle.nn.functional.linear(x, weight, bias)
+            >>> # y: [[0.23824859 0.23824859 0.23824859 0.23824859]
+            >>> #     [0.9440598  0.9440598  0.9440598  0.9440598 ]
+            >>> #     [2.1077576  2.1077576  2.1077576  2.1077576 ]]
```
如无特殊情况，请勿使用 # 的注释方式提供输出值

比如 x , 若要显示输出值，则

```
>>> print(x）
Tensor(shape=[3, 2], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[ 0.06132207,  1.11349595],
        [ 0.41906244, -0.24858207],
        [-1.85169315, -1.50370061]])
```

weight, bias, y 同理。另外，在x代码前记得加上`seed`

sunzhongkai588 commented on python/paddle/nn/functional/extension.py:
```diff
@@ -296,14 +294,14 @@ def gather_tree(ids, parents):
 
-            final_sequences = paddle.nn.functional.gather_tree(ids, parents)
-            # [[[2, 2], [1, 6]], [[3, 3], [6, 1]], [[0, 1], [9, 0]]]
+            >>> final_sequences = paddle.nn.functional.gather_tree(ids, parents)
+            >>> [[[2, 2], [1, 6]], [[3, 3], [6, 1]], [[0, 1], [9, 0]]]
```
同理，如无特殊情况，请显性的提供输出值，如
```suggestion
            >>> final_sequences = paddle.nn.functional.gather_tree(ids, parents)
            >>> print(final_sequences)
            Tensor(shape=[3, 2, 2], dtype=int64, place=Place(cpu), stop_gradient=True,
                   [[[2, 2],
                     [1, 6]],

                    [[3, 3],
                     [6, 1]],

                    [[0, 1],
                     [9, 0]]])
```

sunzhongkai588 commented on python/paddle/nn/initializer/uniform.py:
```diff
@@ -158,24 +158,24 @@ class Uniform(UniformInitializer):
+            ...     name="linear_bias",
+            ...     initializer=paddle.nn.initializer.Uniform(low=-0.5, high=0.5))
+            >>> linear = paddle.nn.Linear(2, 2, weight_attr=weight_attr, bias_attr=bias_attr)
+            >>> # linear.weight:  [[-0.46245047  0.05260676]
+            >>> #                  [ 0.38054508  0.29169726]]
+            >>> # linear.bias:  [-0.2734719   0.23939109]
+
+            >>> res = linear(data)
+            >>> # res:  [[[-0.3553773  0.5836951]]
+            >>> #        [[-0.3553773  0.5836951]]
+            >>> #        [[-0.3553773  0.5836951]]]
```
同理，如无特殊情况，请勿使用 # 的注释方式提供输出值

- 另外，`paddle.nn.initializer.Uniform`有随机性，请在代码开始出加`seed`

```suggestion
            >>> print(linear.weight)
            Parameter containing:
            Tensor(shape=[2, 2], dtype=float32, place=Place(cpu), stop_gradient=False,
                   [[ 0.36583614,  0.02014720],
                    [-0.24039063,  0.40525323]])
                   
            >>> print(linear.bias)
            Parameter containing:
            Tensor(shape=[2], dtype=float32, place=Place(cpu), stop_gradient=False,
                   [-0.07599911, -0.09358713])
                   
            >>> res = linear(data)
            >>> print(res)  
            Tensor(shape=[3, 1, 2], dtype=float32, place=Place(cpu), stop_gradient=False,
                   [[[0.04944640, 0.33181331]],

                    [[0.04944640, 0.33181331]],

                    [[0.04944640, 0.33181331]]])
```

sunzhongkai588 commented on python/paddle/nn/initializer/xavier.py:
```diff
@@ -214,24 +214,24 @@ class XavierNormal(XavierInitializer):
+            ...     name="linear_bias",
+            ...     initializer=paddle.nn.initializer.XavierNormal())
+            >>> linear = paddle.nn.Linear(2, 2, weight_attr=weight_attr, bias_attr=bias_attr)
+            >>> # inear.weight:  [[ 0.06910077 -0.18103665]
+            >>> #                 [-0.02546741 -1.0402188 ]]
+            >>> # linear.bias:  [-0.5012929   0.12418364]
+
+            >>> res = linear(data)
+            >>> # res:  [[[-0.4576595 -1.0970719]]
+            >>> #        [[-0.4576595 -1.0970719]]
+            >>> #        [[-0.4576595 -1.0970719]]]
```
同理，如无特殊情况，请勿使用 # 的注释方式提供输出值，修改方式和上述的 `Uniform` 一样

sunzhongkai588 reviewed (commented):
- 这几个代码示例没过，还是得修改一下

<img width="813" alt="图片" src="https://github.com/PaddlePaddle/Paddle/assets/70642955/84d94b64-8ea2-458c-87e6-0560418aa8b9">

- 另外，还有一些代码的输出结果不要用 `#`，请修改

注意下PR描述里要加上  @megemini @SigureMo @sunzhongkai588 。 改的很不错了，大部份代码示例都通过，加油～

yuchen202 committed:
谢谢孙师傅，我改了一些

 @megemini @SigureMo @sunzhongkai588

megemini commented on python/paddle/nn/functional/common.py:
```diff
@@ -1257,16 +1252,16 @@ def dropout2d(x, p=0.5, training=True, data_format='NCHW', name=None):
+            >>> x = paddle.randn(shape=(2, 3, 4, 5)).astype(paddle.float32)
+            >>> y_train = paddle.nn.functional.dropout2d(x)  #train
+            >>> y_test = paddle.nn.functional.dropout2d(x, training=False) #test
+            >>> for i in range(2):
+            >>>     for j in range(3):
+            >>>         print(x[i,j,:,:])
+            >>>         print(y_train[i,j,:,:]) # may all 0
+            >>>         print(y_test[i,j,:,:])
```
``` python
>>> for i in range(2):
...     for j in range(3):
...         print(x[i,j,:,:])
...         print(y_train[i,j,:,:]) # may all 0
...         print(y_test[i,j,:,:])
```

yuchen202 replied:
请问这个print的结果要放吗

megemini replied:
不用了，for 里面这么多 print ，用注释也行 ～ 看的明白为主 ～

megemini commented on python/paddle/nn/functional/common.py:
```diff
@@ -1821,21 +1822,28 @@ def linear(x, weight, bias=None, name=None):
-          # y: [[0.23824859 0.23824859 0.23824859 0.23824859]
-          #     [0.9440598  0.9440598  0.9440598  0.9440598 ]
-          #     [2.1077576  2.1077576  2.1077576  2.1077576 ]]
+            >>> import paddle
+
+            >>> x = paddle.randn((3, 2), dtype="float32")
+            >>> print(x)
```
加个 seed ～

megemini commented on python/paddle/nn/functional/common.py:
```diff
@@ -2295,12 +2302,12 @@ def fold(
-            # y.shape = [2,3,4,5]
+            >>> x = paddle.randn([2,3*2*2,12])
+            >>> y = F.fold(x, output_sizes=[4, 5], kernel_sizes=2)
+            >>> # y.shape = [2,3,4,5]
```
``` python
>>> print(y.shape)
[2, 3, 4, 5]
```

megemini commented on python/paddle/nn/functional/flash_attention.py:
```diff
@@ -181,13 +181,13 @@ def flash_attention(
+            >>> # required: skiptest
+            >>> import paddle
 
-            q = paddle.rand((1, 128, 2, 16), dtype=paddle.float16)
+            >>> q = paddle.rand((1, 128, 2, 16), dtype=paddle.float16)
 
-            output = paddle.nn.functional.flash_attention(q, q, q, 0.9, False, False)
-            print(output)
+            >>> output = paddle.nn.functional.flash_attention(q, q, q, 0.9, False, False)
+            >>> print(output)
```
1. 这段代码没必要 skip，把旧的 skiptest 去掉。
2. 应该没必要使用 float16, dtype 也可以去掉。
3. flash_attention 没有暴露出去，可以用 paddle.nn.functional.flash_attention.flash_attention 。
4. 这个 output 有点多，可以考虑：
  - 加 seed
  - 改小点 q
  - 打印 output 的其他信息
  - 直接不用 print 了

megemini commented on python/paddle/nn/functional/flash_attention.py:
```diff
@@ -339,13 +339,13 @@ def flash_attn_unpadded(
+            >>> # required: skiptest
+            >>> import paddle
 
-            q = paddle.rand((1, 128, 2, 16), dtype=paddle.float16)
+            >>> q = paddle.rand((1, 128, 2, 16), dtype=paddle.float16)
 
-            output = paddle.nn.functional.flash_attn_unpadded(q, q, q, 0.9, False, False)
-            print(output)
+            >>> output = paddle.nn.functional.flash_attn_unpadded(q, q, q, 0.9, False, False)
+            >>> print(output)
```
同上

megemini commented on python/paddle/nn/initializer/uniform.py:
```diff
@@ -158,24 +158,33 @@ class Uniform(UniformInitializer):
+            ...     name="linear_bias",
+            ...     initializer=paddle.nn.initializer.Uniform(low=-0.5, high=0.5))
+            >>> linear = paddle.nn.Linear(2, 2, weight_attr=weight_attr, bias_attr=bias_attr)
+            >>> print(linear.weight)
```
Uniform 是随机的，需要加个 seed ～

megemini commented on python/paddle/nn/initializer/xavier.py:
```diff
@@ -214,24 +214,33 @@ class XavierNormal(XavierInitializer):
-            # res:  [[[-0.4576595 -1.0970719]]
-            #        [[-0.4576595 -1.0970719]]
-            #        [[-0.4576595 -1.0970719]]]
+            >>> import paddle
+
+            >>> data = paddle.ones(shape=[3, 1, 2], dtype='float32')
+            >>> weight_attr = paddle.framework.ParamAttr(
+            ...     name="linear_weight",
+            ...     initializer=paddle.nn.initializer.XavierNormal())
+            >>> bias_attr = paddle.framework.ParamAttr(
+            ...     name="linear_bias",
+            ...     initializer=paddle.nn.initializer.XavierNormal())
+            >>> linear = paddle.nn.Linear(2, 2, weight_attr=weight_attr, bias_attr=bias_attr)
+            >>> print(linear.weight)
```
同上

megemini commented on python/paddle/nn/initializer/xavier.py:
```diff
@@ -266,24 +275,24 @@ class XavierUniform(XavierInitializer):
-            # res:  [[[ 1.0481861 -2.1206741]]
-            #        [[ 1.0481861 -2.1206741]]
-            #        [[ 1.0481861 -2.1206741]]]
+            >>> import paddle
+
+            >>> data = paddle.ones(shape=[3, 1, 2], dtype='float32')
+            >>> weight_attr = paddle.framework.ParamAttr(
+            ...     name="linear_weight",
+            ...     initializer=paddle.nn.initializer.XavierUniform())
+            >>> bias_attr = paddle.framework.ParamAttr(
+            ...     name="linear_bias",
+            ...     initializer=paddle.nn.initializer.XavierUniform())
+            >>> linear = paddle.nn.Linear(2, 2, weight_attr=weight_attr, bias_attr=bias_attr)
```
同上 ～ 另外，输出 print 一下 ～

yuchen202 committed:
8.29更新

 @megemini @SigureMo @sunzhongkai588
flash_attention.pyflash_attention 没有暴露出去，可以用 paddle.nn.functional.flash_attention.flash_attention 。这个真不会改

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -2081,67 +2178,64 @@ class centers and the shape of sampled_class_center will be [num_positive_class_
+        >>> paddle.seed(2023)
+        >>> label = paddle.randint(low=0, high=num_classes, shape=[batch_size], dtype='int64')
+        >>> remapped_label, sampled_class_index = paddle.nn.functional.class_center_sample(label, num_classes, num_samples)
+        >>> print(label)
+        Tensor(shape=[10], dtype=int64, place=Place(cpu), stop_gradient=True,
+        [10, 3 , 11, 14, 15, 15, 7 , 12, 5 , 13])
+        >>> print(remapped_label)
+        Tensor(shape=[10], dtype=int64, place=Place(cpu), stop_gradient=True,
+        [3, 0, 4, 7, 8, 8, 2, 5, 1, 6])
+        >>> print(sampled_class_index)
+        Tensor(shape=[9], dtype=int64, place=Place(cpu), stop_gradient=True,
+        [3 , 5 , 7 , 10, 11, 12, 13, 14, 15])
```
这里的输出结果都错了，再核对一下
<img width="782" alt="image" src="https://github.com/PaddlePaddle/Paddle/assets/70642955/51f879d3-e2f2-486d-8b37-a2495dbee080">

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -1821,21 +1911,28 @@ def linear(x, weight, bias=None, name=None):
+            >>> print(bias)
+            Tensor(shape=[4], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [1., 1., 1., 1.])
+            >>> y = paddle.nn.functional.linear(x, weight, bias)
```
下面得加输出代码 `print(y)`

sunzhongkai588 commented on python/paddle/nn/functional/activation.py:
```diff
@@ -634,24 +658,24 @@ def rrelu(x, lower=1.0 / 8.0, upper=1.0 / 3.0, training=True, name=None):
+            >>> import paddle
+            >>> import paddle.nn.functional as F
+
+            >>> input_tensor = paddle.to_tensor([[[[-2.0,  3.0, -4.0,  5.0],
```
前面应该加seed，之前review的时候也提到了，`rrelu`是有随机性的

sunzhongkai588 commented on python/paddle/nn/initializer/xavier.py:
```diff
@@ -266,24 +275,32 @@ class XavierUniform(XavierInitializer):
-            #        [[ 1.0481861 -2.1206741]]]
+            >>> import paddle
+
+            >>> data = paddle.ones(shape=[3, 1, 2], dtype='float32')
```
同理，前面加seed，并检查一下输出

yuchen202 committed:
8.30修改

@megemini @SigureMo @sunzhongkai588

sunzhongkai588 commented on python/paddle/nn/functional/common.py:
```diff
@@ -1821,21 +1911,29 @@ def linear(x, weight, bias=None, name=None):
+            [1., 1., 1., 1.])
+            >>> y = paddle.nn.functional.linear(x, weight, bias)
+            >>> print(y)
+            Tensor(shape=[3, 4], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[1.31539762, 1.31539762, 1.31539762, 1.31539762],
+             [0.18737471, 0.18737471, 0.18737471, 0.18737471],
+             [1.23271096, 1.23271096, 1.23271096, 1.23271096]])
```
y的输出再校对一下，好像是错的
<img width="820" alt="image" src="https://github.com/PaddlePaddle/Paddle/assets/70642955/ed9c3da3-b064-4eee-93f9-93ec569748f2">

我测出来也应该是
```
Tensor(shape=[3, 4], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[ 1.58740902,  1.58740902,  1.58740902,  1.58740902],
        [ 1.08524013,  1.08524013,  1.08524013,  1.08524013],
        [-0.67769694, -0.67769694, -0.67769694, -0.67769694]])
```

yuchen202 committed:
8.30下午修改

希望可以合入了

megemini commented on python/paddle/nn/functional/common.py:
```diff
@@ -1257,17 +1252,106 @@ def dropout2d(x, p=0.5, training=True, data_format='NCHW', name=None):
+            >>> paddle.seed(1)
+            >>> x = paddle.randn(shape=(2, 3, 4, 5)).astype(paddle.float32)
+            >>> y_train = paddle.nn.functional.dropout2d(x)  #train
+            >>> y_test = paddle.nn.functional.dropout2d(x, training=False) #test
+            ... for i in range(2):
```
for 前面应该是 `>>> ` 

``` shell
2023-08-30 10:01:40 >>> 1 sample codes ran timeout
2023-08-30 10:01:40 paddle.nn.functional.dropout2d:1 - more than 10.0s
```

错误分析：
1. 因为这里用的是 `...`（既可以是语句，也可以是输出），所以 xdoctest 在 parse docstring 的时候并没有报错（SyntaxError）。
2. 当 xdoctest 实际运行检查的时候，抛出异常（SyntaxError）。
3. 由于目前的 sampcd_processor 用的是 多进程/线程 的方式，所以在 子进程/线程 中抛出异常会卡住。
4. 由于目前的 sampcd_processor 使用了 TIMEOUT 机制，从而在上述异常卡住运行，并超过限制时间的时候，会抛出异常。
5. 最后，此代码被判断为 TIMEOUT，与 CI 的结果一致。

如果要捕获 run 时候的异常，得要重新设计，感觉没啥必要，毕竟有 TIMEOUT 这个机制做保护，错误信息目前也能从日志中看到 ～

to @megemini 记录一下，后续注意～

yuchen202 replied:
所以改一下是吗

yuchen202 committed:
... for i in range(2):

将...改为>>>

sunzhongkai588 reviewed (approved):
LGTM for docs
